modules/agentless-s3-bucket/variables.tf

```diff
@@ -0,0 +1,26 @@
+
+variable "tags" {
+  description = "A map of additional tags to add to the IAM role/profile created"
+  type        = map(string)
+  default     = {}
+}
+
+variable "iam_delegate_role_arn" {
+  description = "Reference ARN to the Agentless delegate role"
+  type        = string
+}
+
+variable "iam_rds_assume_role_arn" {
+  description = "Reference ARN to the Agentless RDS assume role"
+  type        = string
+}
+
+variable "primary_kms_key_arn" {
```

Comment:
why 'primary'? is there a secondary?

Reply:
We create a multi-region kms key with the delegate role which is the primary one and we create replicas for each region where the s3 module is deployed.
We need a KMS Key in the same region as the bucket and the RDS snapshot.
Given these constraints we had a couple of options: we could create a key for each region we deploy the bucket to or we create a multi-region kms key, with a replica in each of the rds scanning regions.
We decided to have the multi-region kms key approach for the following regions:
- (not very important) we only pay for one kms key (the primary) instead of multiple keys
- we need to allow the delegate role to access the keys. If we create the keys in this module, we are making a circular dependency between the two modules.